app/src/main/java/fi/bitrite/android/ws/ui/FilterListFragment.java

```diff
@@ -0,0 +1,152 @@
+        mLastAccessSeekBar.setOnSeekBarChangeListener(this);
+        mLastAccessSeekBar.setProgress(convertLastAccessDaysToProgress(
+              mHostFilterManager.getFilterValue(HostFilterManager.RECENT_ACTIVITY_FILTER_KEY)));
+        mLastAccessSeekBar.setMax(10);
+
+        mCurrentlyAvailableCheckBox.setChecked(
+            mHostFilterManager.isFilterActive(HostFilterManager.CURRENTLY_AVAILABLE_FILTER_KEY));
+        return view;
+    }
+
+    @OnClick(R.id.currently_available_checkbox)
+    void onCurrentlyAvailableClicked(View view) {
+        boolean checked = ((CheckBox) view).isChecked();
+        if (checked) {
+            mHostFilterManager.activateFilter(HostFilterManager.CURRENTLY_AVAILABLE_FILTER_KEY);
+        } else {
+            mHostFilterManager.deactivateFilter(HostFilterManager.CURRENTLY_AVAILABLE_FILTER_KEY);
+        }
+    }
+
+    @OnClick(R.id.btn_clear_filters)
+    void onClearFiltersClicked(View view) {
+        mLastAccessSeekBar.setProgress(10);
+        mCurrentlyAvailableCheckBox.setChecked(false);
+        mHostFilterManager.deactivateFilter(HostFilterManager.CURRENTLY_AVAILABLE_FILTER_KEY);
+    }
+
+    @Override
+    protected CharSequence getTitle() {
+        return getString(R.string.title_fragment_filter_list);
+    }
+
+    @Override
+    public void onProgressChanged(SeekBar seekBar, int progress, boolean fromUser) {
+      mLastAccessText.setText(convertLastAccessProgressToText(progress));
+      mHostFilterManager.updateFilterValue(HostFilterManager.RECENT_ACTIVITY_FILTER_KEY,
+          convertLastAccessProgressToDays(progress));
+    }
+    @Override
+    public void onStartTrackingTouch(SeekBar seekBar) { }
+    @Override
+    public void onStopTrackingTouch(SeekBar seekBar) { }
+
+    // [0, 1, ... 10] -> {1, 2, 3, 7, 14, 30, 60, 90, 180, 365, -1}.
+    private int convertLastAccessProgressToDays(int progress) {
+        switch (progress) {
+            case 0: return 1;
+            case 1: return 2;
+            case 2: return 3;
+            case 3: return 7;
+            case 4: return 14;
+            case 5: return 30;
+            case 6: return 60;
+            case 7: return 90;
+            case 8: return 180;
+            case 9: return 365;
+            case 10: return -1;
+        }
+        return -1;
+    }
+
+    // Inversion of convertLastAccessProgressToDays.
+    private int convertLastAccessDaysToProgress(int days) {
+        switch (days) {
+            case 1: return 0;
+            case 2: return 1;
+            case 3: return 2;
+            case 7: return 3;
+            case 14: return 4;
+            case 30: return 5;
+            case 60: return 6;
+            case 90: return 7;
+            case 180: return 8;
+            case 365: return 9;
+            case -1: return 10;
+        }
+        return 10;
+    }
+
+    // TODO(dproctor): This needs to be internationalized. Perhaps something in
+    // android.text.format.DateUtils can be used?
```

Comment:
Yes, probably one of the implementations of [ `DateUtils.getRelativeTimeString()`](https://developer.android.com/reference/android/text/format/DateUtils.html?hl=es#getRelativeDateTimeString(android.content.Context,%20long,%20long,%20long,%20int))

Reply:
Ugh, unfortunately DateUtils.getRelativeTimeSpanString() _almost_ does what I want, but unfortunately, it won't do anything more than "xx weeks ago" (or will just give the date for a min resolution of a day, which is confusing), so special casing will be required anyways, as I think the current behavior is considerably more usable than anything that DateUtils will generate. Seems best just to bit the bullet and add a few more string resources :/